### PR TITLE
Change oath token to header & call order when intializing list request

### PR DESCRIPTION
**Pull request by aalexx-S**
Change access token to use authorization header instead of oath token.

Change the call order for the initializeRequest method and setting the
object prefix for object list reuqest.

### CODE REVIEW

**Review by cyxxy — APPROVED** [review]

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java [hunk 1]
```diff
@@ -1496,6 +1495,8 @@ private String listStorageObjectsAndPrefixesPage(
       listObject.setPrefix(objectNamePrefix);
     }
 
+    listObject = initializeRequest(listObject, bucketName);
```
Comment (medb) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java:1498]: What information `initializeRequest` needs that we can not call it at the very beginning?

Comment (aalexx-S) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java:1498]: The set prefix call at L:1495. Without setting prefix, the request looks like a list at bucket root level.

Comment (medb) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java:1498]: Can we inline it in the `initializeRequest` then?

```java
Storage.Objects.List listObject =
        initializeRequest(
                storage.objects().list(bucketName).setPrefix(emptyToNull(objectNamePrefix)),
                bucketName);
```

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java [hunk 2]
```diff
@@ -2198,13 +2196,27 @@ public GoogleCloudStorageItemInfo composeObjects(
     return compositeInfo;
   }
 
+  /**
+   * Initialize http request.
+   *
+   * <p>Initialize the storage access requests. Since access request authorization are done here and
+   * right before sending out the http requests, the passed in storage requests should contain as
```
Comment (medb) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java:2203]: nit: s/http/HTTP

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageTest.java [hunk 3]
```diff
@@ -3220,7 +3220,8 @@ public void testIgnoreExceptionsOnCreateEmptyObject() throws IOException {
 
     gcs.createEmptyObject(
         RESOURCE_ID,
-        CreateObjectOptions.DEFAULT_OVERWRITE.toBuilder()
+        CreateObjectOptions.DEFAULT_OVERWRITE
```
Comment (medb) [inline gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageTest.java:3223]: Is this `google-java-format` change?

**Review by medb — CHANGES_REQUESTED** [review]

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageTest.java (hunk 3 above)
Comment (aalexx-S) [reply gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageTest.java:3223]: Idk, there is a maven build check for coding style and it is complaining about this

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java (hunk 2 above)
Comment (aalexx-S) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java]: Done

**Review by medb — APPROVED** [review]

**Pull request merged**